### ISSUE TITLE
Issues compiling facebook/pfff within homebrew formula

### ISSUE THREAD

**Issue by molovo** [Homebrew/homebrew#46061]
I'm trying to write a formula for facebook's [pfff](http://github.com/facebook/pfff) tools. If I clone the HEAD and follow the build steps in a normal shell everything compiles fine, but within the Homebrew environment I get `clang: error: no input files` errors appearing, once during `configure`, and multiple times during `make depend`. I've tried it both with the formula, and in the interactive install shell.

I'm reasonably new to writing Formulae, so I may be missing something, but as I said everything compiles without any errors when using exactly the same commands outside of the homebrew environment.

Gist logs: <https://gist.github.com/anonymous/9b73910ed17c19e428cb>

Formula:
```ruby
class Pfff < Formula
  desc "Tools for code analysis, visualizations, or style-preserving source transformation."
  homepage "http://github.com/facebook/pfff"
  head "https://github.com/facebook/pfff.git"

  depends_on "objective-caml"
  depends_on "camlp4"
  depends_on "gtk+"
  depends_on "cairo"
  depends_on "homebrew/x11/swi-prolog"

  env :userpaths

  def install
    args = %W[
      --visual
      --bytecode
      --prefix=#{prefix}/bin
    ]

    (etc/"pfff").install Dir["*"]

    (prefix/"bin").mkpath

    cd etc/"pfff" do
      system "./configure", *args
      system "make depend"
      system "make"
      system "make opt"
    end

    cd prefix/"bin" do
      bin.install_symlink Dir["*"]
    end
  end
end
```

**Comment by molovo** [comment]
Managed to get round the issue with `configure` and `make depend` by forcing `/usr/bin` to the start of `$PATH` so that the system clang is used. Despite them being the same version that allows the first two steps to pass. (Don't know if there's a more accepted way of doing this, but it worked).

I'm now running into another issue with `make` (the 3rd step). Not entirely sure if this is a Homebrew issue, or an issue with the library, but again, *this step passes outside of the Homebrew environment*.

New Gist Logs: <https://gist.github.com/39f5427775ea270d6437>

Updated formula:
```ruby
class Pfff < Formula
  desc "Tools for code analysis, visualizations, or style-preserving source transformation."
  homepage "http://github.com/facebook/pfff"
  head "https://github.com/facebook/pfff.git"

  depends_on "objective-caml"
  depends_on "camlp4"
  depends_on "gtk+"
  depends_on "cairo"
  depends_on "homebrew/x11/swi-prolog"

  env :userpaths

  def install
    ENV["PATH"] = "/usr/bin:#{ENV["PATH"]}"

    prefix.install Dir["*"]

    args = %W[
      --visual
      --bytecode
    ]

    cd prefix do
      system "./configure", *args
      system "make depend"
      system "make"
      system "make opt"

      bin.install_symlink "codegraph"
      bin.install_symlink "codemap"
      bin.install_symlink "codequery"
      bin.install_symlink "codeslicer"
      bin.install_symlink "pfff"
      bin.install_symlink "pfff_db"
      bin.install_symlink "pfff_test"
      bin.install_symlink "scheck"
      bin.install_symlink "sgrep"
      bin.install_symlink "spatch"
      bin.install_symlink "stags"
    end
  end
end
```

**Comment by aryx** [comment]
I think it is because of the MAKEFLAGS: -j8
It does things in parallel where it should not.
Can you change that in the formula?
Otherwise I'll try to fix it in pfff/.

**Comment by aryx** [comment]
A quick fix is   make -j1 I guess.

**Comment by molovo** [comment]
That's fixed it. Cheers @aryx
